OmniKit/MessageTransport/MessageBlocks/PodInfoFaultEvent.swift

```diff
@@ -10,118 +10,106 @@ import Foundation
-        self.recieverLowGain = encodedData[18] >> 4
+        self.radioRSSI =  Int8(encodedData[18] & 0x3F)
         
-        self.radioRSSI =  encodedData[18] & 0xF
-        
-        guard let reservoirStatusAtFirstLoggedFaultEventCheckType = ReservoirStatus(rawValue: encodedData[19] & 0xF) else {
-            throw MessageError.unknownValue(value: encodedData[19] & 0xF, typeDescription: "ReservoirStatus")
+        guard let previousPodProgressStatus = PodProgressStatus(rawValue: encodedData[19] & 0xF) else {
+            throw MessageError.unknownValue(value: encodedData[19] & 0xF, typeDescription: "PodProgressStatus")
         }
-        self.reservoirStatusAtFirstLoggedFaultEventCheck = reservoirStatusAtFirstLoggedFaultEventCheckType
+        self.previousPodProgressStatus = previousPodProgressStatus
+        
+        self.unKnownValue = encodedData[20...21]
         
         self.data = Data(encodedData)
     }
 }
 
 extension PodInfoFaultEvent: CustomDebugStringConvertible {
     public typealias RawValue = Data
-
     public var debugDescription: String {
         return [
             "## PodInfoFaultEvent",
             "rawHex: \(data.hexadecimalString)",
-            "originalLoggedFaultEvent: \(originalLoggedFaultEvent)",
-            "secondaryLoggedFaultEvent: \(secondaryLoggedFaultEvent)",
-            "reservoirStatus: \(reservoirStatus)",
-            "deliveryType: \(deliveryType)",
-            "insulinNotDelivered: \(insulinNotDelivered)",
-            "unknownPageCode: \(unknownPageCode)",
-            "faultEventTimeSinceActivation: \(faultEventTimeSinceActivation)",
-            "insulinRemaining: \(insulinRemaining)",
-            "timeActive: \(timeActive)",
+            "currentStatus: \(currentStatus.description)",
+            "previousStatus: \(previousStatus.description)",
+            "podProgressStatus: \(podProgressStatus)",
+            "deliveryType: \(deliveryType.description)",
+            "podProgressStatus: \(podProgressStatus)",
+            "reservoirLevel: \(String(describing: reservoirLevel)) U",
+            "timeActive: \(timeActive.stringValue)",
             "logEventError: \(logEventError)",
-            "reservoirStatusAtFirstLoggedFaultEvent: \(reservoirStatusAtFirstLoggedFaultEvent)",
-            "recieverLowGain: \(recieverLowGain)",
+            "logEventErrorType: \(logEventErrorType.description)",
+            "logEventErrorPodProgressStatus: \(logEventErrorPodProgressStatus)",
+            "recieverLowGain: \(receiverLowGain)",
             "radioRSSI: \(radioRSSI)",
-            "reservoirStatusAtFirstLoggedFaultEventCheck: \(reservoirStatusAtFirstLoggedFaultEventCheck)",
-            "insulinStateTableCorruption: \(insulinStateTableCorruption)",
-            "immediateBolusInProgress: \(immediateBolusInProgress)",
+            "recieverLowGain: \(receiverLowGain)",
+            "previousPodProgressStatus: \(previousPodProgressStatus)",
+            "unKnownValue: \(unKnownValue)",
```

Comment:
This is a Data object right? Do you want `unKnownValue.hexadecimalString`?

Reply:
Ah yes, that's what's needed for now.